src/cider/nrepl/middleware/util/storage.clj

```diff
@@ -0,0 +1,41 @@
+(ns cider.nrepl.middleware.util.storage)
+
+(def ^:private store (ref {}))
+(def ^:private exp-index (ref '()))
+(def ^:private capacity 10)
+
+(defn add!
+  "Adds the provided kv pair to a single-retrieval store with
+  queue-capacity expiring behavior; i.e. not only is kv removed from
+  the store when sucessfully queried, but if addition of the supplied
+  item causes store size to increase over the default capacity of 10
+  items, then the oldest kv pair in the store is also
+  expired (removed). Returns the value added to the store."
+  [k v]
+  (dosync
```

Comment:
Is it just me, or is this caching code more complex than it needs to be. Btw, I guess you're aware of https://github.com/clojure/core.cache